modules/account/src/main/kotlin/edu/stanford/spezi/module/account/manager/AuthenticationManager.kt

```diff
@@ -49,69 +57,105 @@ internal class AuthenticationManager @Inject constructor(
+                .await().let { }
         }.onFailure { e ->
             logger.e { "Error saving user data: ${e.message}" }
-        }.isSuccess
+        }
     }
 
     suspend fun sendForgotPasswordEmail(email: String): Result<Unit> {
         return runCatching {
-            firebaseAuth.sendPasswordResetEmail(email).await()
-            Unit
+            firebaseAuth.sendPasswordResetEmail(email).await().let { }
         }.onFailure { e ->
             logger.e { "Error sending forgot password email: ${e.message}" }
         }
     }
 
-    suspend fun signInWithEmailAndPassword(email: String, password: String): Boolean {
+    suspend fun signIn(email: String, password: String): Result<Unit> {
         return runCatching {
             val result = firebaseAuth.signInWithEmailAndPassword(email, password).await()
-            result.user != null
+            if (result.user == null) error("Failed to sign in, returned null user")
         }.onFailure { e ->
             logger.e { "Error signing in with email and password: ${e.message}" }
-        }.getOrDefault(false)
+        }
     }
 
-    suspend fun signInWithGoogle(idToken: String): Result<Boolean> {
+    suspend fun signInWithGoogle(): Result<Unit> {
         return runCatching {
-            val credential = GoogleAuthProvider.getCredential(idToken, null)
+            val googleIdTokenCredential = getCredential(filterByAuthorizedAccounts = true) ?: error("Failed to get credential")
+            val credential = GoogleAuthProvider.getCredential(googleIdTokenCredential.idToken, null)
             logger.i { "Credential: $credential" }
             val result = firebaseAuth.signInWithCredential(credential).await()
             logger.i { "Result: $result" }
-            result.user != null
+            if (result.user == null) error("Failed to sign in, returned null user")
         }.onFailure {
             logger.e { "Error signing in with google: ${it.message}" }
         }
     }
+
+    suspend fun getCredential(filterByAuthorizedAccounts: Boolean): GoogleIdTokenCredential? {
+        return runCatching {
+            val googleIdOption = GetGoogleIdOption.Builder()
+                .setFilterByAuthorizedAccounts(filterByAuthorizedAccounts)
+                .setAutoSelectEnabled(true)
+                // TODO: Uncomment once secrets xml has been added in CI secrets
+                // .setServerClientId(context.getString(R.string.serverClientId))
+                .build()
+
+            val request = GetCredentialRequest.Builder()
+                .addCredentialOption(googleIdOption)
+                .build()
+
+            val credential = credentialManager.getCredential(
+                request = request,
+                context = context
+            ).credential
+            if (credential is CustomCredential && credential.type == GoogleIdTokenCredential.TYPE_GOOGLE_ID_TOKEN_CREDENTIAL) {
+                GoogleIdTokenCredential.createFrom(credential.data)
```

Comment:
This method is not tested because of this usage, mockk could not stub the static behaviour on this component